src/httpserver/http_request.hpp

```diff
@@ -442,7 +444,7 @@ class http_request
         **/
         void set_arg(const std::string& key, const std::string& value)
         {
-            this->args[key] = value;
+            this->args[key] = value.substr(0,content_size_limit);
```

Comment:
In this case, it feels like throwing an exception would be more transparent - at least the library consumer will be notified that it happened and will avoid using partial data as argument unless they want to.